plugin.video.piped/lib/services/entrypoint.py

```diff
@@ -0,0 +1,28 @@
+from threading import Thread
+import xbmc
+
+from lib.services.httpserver import HttpService
+from lib.services.scheduler import Scheduler
+from lib.services.player import Player
+
+class Service(xbmc.Monitor):
+	def __init__(self):
+		httpservice = Thread(target=HttpService)
+		scheduler = Scheduler()
+		httpservice.daemon = True
+		scheduler.daemon = True
+		player = Player()
+		player.isPlaying()
+
+		while not self.abortRequested():
+			if self.waitForAbort(5):
+				while scheduler.is_alive():
+					scheduler.stop()
+					xbmc.sleep(200)
```

Comment:
If `scheduler.stop()` does not finish, the script waits for 200ms and then stops it again? In the `Scheduler.stop()` you call `self.join()` which should actually block until the thread stopped.